org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BuildWorkspaceHandler.java

```diff
@@ -61,8 +60,14 @@ public BuildWorkspaceStatus buildWorkspace(boolean forceReBuild, IProgressMonito
 					connection.publishDiagnostics(new PublishDiagnosticsParams(ResourceUtils.toClientUri(uri), Collections.emptyList()));
 				}
 			}
-			ResourcesPlugin.getWorkspace().build(forceReBuild ? IncrementalProjectBuilder.FULL_BUILD : IncrementalProjectBuilder.INCREMENTAL_BUILD, monitor);
-			List<IMarker> problemMarkers = workspaceDiagnosticsHandler.publishDiagnostics(monitor);
+			ResourcesPlugin.getWorkspace().build(forceReBuild ? IncrementalProjectBuilder.CLEAN_BUILD : IncrementalProjectBuilder.INCREMENTAL_BUILD, monitor);
+			List<IMarker> problemMarkers = new ArrayList<>();
+			for (IProject project : projects) {
```

Comment:
should ignore default project

Reply:
BuildWorkspaceHandlerTest.testFailedCase() fails.

Reply:
mmok